services/registration/service/mail_service.go

```diff
@@ -30,3 +30,37 @@ func SendUserMail(id string, template string) error {
 
 	return err
 }
+
+/*
+	Add user with given id, to the specified mailing list.
+	If the mailing list does not exist, it creates a new list with the user.
+*/
+func AddUserToMailList(user_id string, mail_list_id string) error {
+	add_to_mail_list_url := fmt.Sprintf("%s/mail/list/add/", config.MAIL_SERVICE)
+
+	mail_list := models.MailList{
+		ID:      mail_list_id,
+		UserIDs: []string{user_id},
+	}
+
+	update_request_body := bytes.Buffer{}
+	json.NewEncoder(&update_request_body).Encode(&mail_list)
+
+	content_type := "application/json"
+
+	_, err := http.Post(add_to_mail_list_url, content_type, &update_request_body)
+
+	if err != nil {
```

Comment:
If the mailing list doesn't exist err will still be nil since the request happened without error. You'll need to check the code on the response to see if the user was successfully added to the list.